egui/src/localization.rs

```diff
@@ -0,0 +1,100 @@
+//! Localization of texts in widgets and containers.
+use std::default::Default;
+
+/// Handles the localization of default texts in [`crate::widgets`] and [`crate::containers`].
+///
+/// You can set the current language with [`crate::Context::set_localization`]. For example: `ctx.set_localization(Language::BahasaMalaysia)`.
+#[derive(Clone, Debug)]
+#[cfg_attr(feature = "persistence", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "persistence", serde(default))]
+pub struct Localization {
+    /// The current language used for texts.
+    pub lang: Language,
+
+    // Texts for sliders
+    pub slider_tooltip: &'static str,
+
+    // Texts for colour pickers
+    pub click_copy: &'static str,
+    pub cp_edit: &'static str,
+    pub cp_blending: &'static str,
+    pub cp_additive: &'static str,
+    pub cp_normal: &'static str,
+    pub cp_selected_color: &'static str,
+    pub cp_hue: &'static str,
+    pub cp_saturation: &'static str,
+    pub cp_value: &'static str,
+    pub lang_text: &'static str,
+}
+
+impl Default for Localization {
+    /// Sets English as the default language for texts.
+    ///
+    /// It can also be used to switch from another language to English.
+    fn default() -> Self {
+        Self {
+            lang: Language::English,
+            slider_tooltip: "Drag to edit or click to enter a value.\nPress 'Shift' while dragging for better control",
+            click_copy: "Click to copy",
+            cp_edit: "Click to edit color",
+            cp_blending: "Blending",
+            cp_additive: "Additive",
+            cp_normal: "Normal",
+            cp_selected_color: "Selected color",
+            cp_hue: "Hue",
+            cp_saturation: "Saturation",
+            cp_value: "Value",
+            lang_text: "Language",
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(feature = "persistence", derive(serde::Deserialize, serde::Serialize))]
+/// Specifies the languages currently available for localization and is required by [`crate::Context::set_localization`] as the parameter type.
+pub enum Language {
+    English,
+    BahasaMalaysia,
```

Comment:
Do we need this list? Can't the user just call `context.set_localization(Localization::malay());` instead? That way it is also easy for a user to add a new language without having to make a PR to egui.

Reply:
`Localization::lang` could be changed to `&'static str` too (e.g. "english")

Reply:
> `Localization::lang` could be changed to `&'static str` too (e.g. "english")

Agree, some clarification: `<language code>[_<territory code>]` should be used (`en`, `en_US` and so on)